code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -2298,6 +2308,13 @@ obj/item/device/pda/AltClick()
 		note = A:info
 		to_chat(user, "<span class='notice'>Paper scanned.</span>")//concept of scanning paper copyright brainoblivion 2009
 
+/obj/item/device/pda/afterattack(atom/A as mob|obj|turf|area, mob/user as mob)
```

Comment:
The `as x` stuff does nothing, remove it.

Reply:
Also, `afterattack` has a third parameter called `proximity_flag`. It's true when the user is adjacent, so you can use that to replace the `A.Adjacent(user)` down below.